chromium_src/chrome/browser/ui/views/tabs/tab_container.cc

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/tabs/brave_tab_container.h"
+#include "brave/browser/ui/views/tabs/brave_vertical_tab_utils.h"
+
+#include "src/chrome/browser/ui/views/tabs/tab_container.cc"
+
+// In order to instantiate RemoveTabDelegate which is defined in
+// tab_container.cc, this method is defined here.
+void BraveTabContainer::StartRemoveTabAnimation(Tab* tab,
+                                                int former_model_index) {
+  if (!tabs::ShouldShowVerticalTabs()) {
+    TabContainer::StartRemoveTabAnimation(tab, former_model_index);
+    return;
+  }
+
+  if (in_tab_close_ && GetTabCount() > 0 &&
+      override_available_width_for_tabs_ >
+          tabs_view_model_.ideal_bounds(GetTabCount() - 1).right()) {
+    // Tab closing mode is no longer constraining tab widths - they're at full
+    // size. Exit tab closing mode so that it doesn't artificially inflate our
+    // bounds.
+    ExitTabClosingMode();
+  }
+
+  StartBasicAnimation();
+
+  const int tab_overlap = TabStyle::GetTabOverlap();
+
+  // TODO(pkasting): When closing multiple tabs, we get repeated RemoveTabAt()
```

Comment:
Please create a tracking issue for this, but we should follow the behavior from the normal tab strip